File: src/main/java/de/byjoker/myjfql/user/session/SessionService.java
package de.byjoker.myjfql.user.session;

import de.byjoker.myjfql.util.StorageService;

import java.util.List;

public interface SessionService extends StorageService {

    void openSession(Session session);

    void saveSession(Session session);

    void closeSession(String token);

    void closeSessions(String userId);

    void collectExpiresSessions();

    boolean existsSession(String token);

    Session getSession(String token);

    List<Session> getSessionsByUserId(String userId);

    List<Session> getSessions();

}

File: src/main/java/de/byjoker/myjfql/network/session/SessionService.kt
package de.byjoker.myjfql.network.session

interface SessionService {

    fun openSession(session: Session)

    fun closeSession(token: String)

    fun closeSessions(userId: String)

    fun saveSession(session: Session)

    fun getSession(token: String): Session?

    fun getSessionsByUserId(userId: String): List<Session>

    val sessions: MutableList<Session>

}
